orttraining/orttraining/models/runner/training_runner.h

```diff
@@ -55,6 +57,10 @@ class TrainingRunner {
 
     MapStringToString input_name_map;
 
+    // used for collecting perf metrics
+    std::map<std::string, std::pair<std::string, size_t>> metrics_map;
```

Comment:
it's not obvious from the name and comment here how to use these.
looks like they are used to determine the sequence length and predictions per sequence from the input data. they seem generic (e.g., can specify some mapping or store arbitrary properties) but in fact TrainingRunner::SavePerfMetrics() looks for specific properties.

one idea, instead of computing this model-specific data from within TrainingRunner, can we just give it arbitrary model-specific data to put into the perf metrics? e.g., sequence length and predictions per sequence could be computed outside (in main?) and passed to TrainingRunner

Reply:
will check ...